app/upgrades/signetlaunch/data_signed_fps.go

```diff
@@ -0,0 +1,122 @@
+package signetlaunch
+
+const SignedFPsStr = `{
```

Comment:
Is it a hard requirement that the string needs to contain the full tx rather
than just the message part?

Reply:
I think I will attach myself to this comments as this is probably closest to my concerns.

I think the question here becomes where do we validates `bbn1gwecky0m842kvjg7mcvt9z330rz96948aplqlm` is if fact the signer of the transactions:

1. We can do it off-chain, in some process /service and here include only the messages ?
2. W can do it on-chain, so gather transactions to some database and during upgrade perform whole transaction validation including transaction signaure. To do that we would need to move all of those transaction through message router as transactions signature validation is performed in `anteHandler`. Then we would ignore the ones that fail the sig validation (though this seems like bad ux ?)

Reply:
yes, we can do it off-chain as option 1. 
or in option 2. I am validating during the upgrade https://github.com/babylonlabs-io/babylon/blob/5215030b9b8e4d51fb63a633cc531c7e4f445224/app/upgrades/signetlaunch/upgrades.go#L177-L178

Reply:
Do we really need to validate the signature of the tx signed by the signer?

It's ourselves who determine `SignedFPsStr` and we will need to gather data and verify them offline anyway. Only after we are confident `SignedFPsStr` is correct we will perform this software upgrade.

So it seems that inserting messages directly is already secure?

Reply:
Updated at https://github.com/babylonlabs-io/babylon/pull/28/commits/79a07120715aa0096c7ddf9c2537721f5627eea5

One thing to note is, that I would still expect finality providers to give us the entire signed message during the registry and we convert it to the Finality provider information as needed 

Mostly as a guarantee that the fp owns that key and is confirming that he wants to be a FP for that BTC pk, now that the PoP only covers BTC owns address :sweat_smile:

Reply:
what misled me, is that the line you linked before ie
```
cdc.InterfaceRegistry().SigningContext().AddressCodec().BytesToString(signers[0])
```
does not really verify signature. To verify signature easiest way would be copy code from ante handler:
https://github.com/cosmos/cosmos-sdk/blob/main/x/auth/ante/sigverify.go#L302

> One thing to note is, that I would still expect finality providers to give us the entire signed message during the registry and we convert it to the Finality provider information as needed
> 
> Mostly as a guarantee that the fp owns that key and is confirming that he wants to be a FP for that BTC pk, now that the PoP only covers BTC owns babylon address

totally agree on that we need to verify signature for provider Babylon address. 

One things that worries me, if we miss something and add transactions with invalid signature to this string, and now we won't validate it on node level. It seems risky.  Maybe, as this whole string is part of our binary, we can have unit test that goes over whole thing and verify signatures ?

Reply:
Would You prefer to have the full signed message in the string and validate the signature during the upgrade? 
Or validate it offline and simplify the upgrade with just the Finality Provider information? 

About the `sigverify`: 
this improvement in `sigverify` is still unreleased https://github.com/cosmos/cosmos-sdk/blob/0fe3115dcf2e42dac29364a70cca71d52df2a443/x/auth/CHANGELOG.md#L38

The one we currently have loads the account from the AccountKeeper 
and this is not good for us, since the account was not created yet by not submitting any transaction by the finality provider babylon key holder before

![image](https://github.com/user-attachments/assets/f81e09b9-96de-480f-8299-d891085c868c)

Reply:
As discussed with @KonradStaniec offline, the string with finality providers will contain the entire transaction with signature, but the signatures validity, PoP and other checks will not be verified during the upgrade, only by a unit test `TestValidateSignatureSignedFPsFromData` that should make sure everything is checked for the upgrade to happen on that front

So, during the upgrade we would still need to parse the msg to `Any` to be able to cast it to the `MsgCreateFinalityProvider` after, but without validating other things

cc @SebastianElvis

Reply:
I will wait for you guys approval on this before merge